packages/validator/test/unit/utils/clock.test.ts

```diff
@@ -84,6 +84,14 @@ describe("util / Clock", function () {
     // eslint-disable-next-line @typescript-eslint/naming-convention
     const testConfig = {SECONDS_PER_SLOT: 12} as BeaconConfig;
     const genesisTime = Math.floor(new Date("2021-01-01").getTime() / 1000);
+
+    // Tests can fail under certain time slots, overriding the system time 
+    // with a specific value allows us to run tests deterministically
+    // See : https://github.com/ChainSafe/lodestar/issues/6734
+    beforeEach(() => {
+      vi.setSystemTime(genesisTime * 1000);
+    });
```

Comment:
We are using `vi.useFakeTimers({now: Date.now()});` that config `now` is enough to make the tests deterministic. If any regression detected we should try to hardcode value `now` to a static value.

Reply:
> that config now is enough to make the tests deterministic

but the test is not deterministic, see https://github.com/ChainSafe/lodestar/issues/6734#issuecomment-2128367331

Reply:
> We are using `vi.useFakeTimers({now: Date.now()});` that config `now` is enough to make the tests deterministic. If any regression detected we should try to hardcode value `now` to a static value.

We need to set current time to be proportionate to `genesisTime` in order to make this particular test deterministic

Reply:
Another [failed run](https://github.com/ChainSafe/lodestar/actions/runs/9420498728/job/25952845970#step:5:1644) on unstable, we should merge this